### ISSUE TITLE
Pty host becomes unresponsive when creating a new terminal right after manually relaunching an existing one

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#121336]
Related to https://github.com/microsoft/vscode/issues/117956

Repro on Windows:

1. Open a terminal
2. Type something into it
3. Toggle the auto attach value, wait for the ⚠️ icon in the terminal
4. Relaunch the terminal by hovering ⚠️ 
5. Create a new terminal

![image](https://user-images.githubusercontent.com/2193314/114792393-27ba0380-9d3d-11eb-9b24-16fa83370b9a.png)

**Comment by Tyriar** [comment]
Shared process log when this happens:

```
[IPC Library: Pty Host] TRACE IPty#spawn C:\Program Files\PowerShell\7\pwsh.exe [] {name: "pwsh.exe", cwd: "E:\GitHub\microsoft\vscode-hexeditor", env: {…}, cols: 58, rows: 16, …}
console.ts:137 [IPC Library: Pty Host] TRACE IPty#kill
WARN No ptyHost heartbeat after 6 seconds
log.ts:296   ERR No ptyHost heartbeat after 12 seconds
```

**Comment by Tyriar** [comment]
I'm beginning to think this is just another manifestation of https://github.com/microsoft/vscode/issues/76548

**Comment by Tyriar** [comment]
Yep, the fix for that also fixes this

**Comment by meganrogge** [comment]
This actually reproes after any manual relaunch for me. Just takes 10 seconds
![recording - 2021-04-23T132258 708](https://user-images.githubusercontent.com/29464607/115926110-1d7fbf80-a437-11eb-8596-1e760b13bdb5.gif)

**Comment by Tyriar** [comment]
Things I tried:

- Throttling spawn calls
- Throttling kill calls
- Throttling both spawn and kill calls, ensuring they don't happen near each other
- Ensure no kill call happens within seconds of process data events

Nothing seemed to fix it. Pushing this to May, I don't think investing more in a mitigation is worth the effort here, we should work on the actual asar fix.

**Comment by gganjang** [comment]
Re-installing(removed VSCode from windows setting and downloaded stable version(1.56.0) from the website) VSCode worked for me

**Comment by markserlin-crucible** [comment]
I am seeing this issue repeatedly
Version: 1.55.2 (user setup)
Commit: 3c4e3df
Date: 2021-04-13T09:35:57.887Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.19041

When > 1 terminal window is open either side by side or in separate mode, or indeed on sepearte instances of VSCode.
PS I use the Terminal confgured as a git bash

I believe issue was closed for 1.54.x but see version above

I tried the latest Insider's version and the Terminal dod not open at all - completely broken.
Steps: Menu->Terminal->New Terminal
No response.
